core/services/src/main/java/org/exoplatform/services/attachments/service/AttachmentServiceImpl.java

```diff
@@ -211,14 +250,18 @@ public void deleteAttachmentItemById(long userIdentityId,
       throw new IllegalAccessException("User with name " + userIdentityId + " doesn't exist");
     }
 
-    AttachmentContextEntity attachmentContextEntity = attachmentStorage.getAttachmentItemByEntity(entityId,
-                                                                                                  entityType,
-                                                                                                  attachmentId);
-    if (attachmentContextEntity == null) {
+    Attachment attachmentEntity = null;
+    try {
+      attachmentEntity = attachmentStorage.getAttachmentItemByEntity(entityId, entityType, attachmentId);
+    } catch (Exception e) {
+      throw new NotFoundException("Can't convert attachment JCR node to Attachment DTO", e);
```

Comment:
why using Clouddrive exception ?

Reply:
I suggest to avoid catching the exception here and let upper layer receive and handle the exception, unless you have an important information to include in exception that will help, diagnose the original problem on runtime (which is not the case, because the message is a static string input and no variables or additional information is added)